src/MongoDB.Bson/IO/BsonBinaryReader.cs

```diff
@@ -148,16 +148,25 @@ public override BsonBinaryData ReadBinaryData()
 
             var bytes = _bsonStream.ReadBytes(size);
 
-            GuidRepresentation guidRepresentation;
-            switch (subType)
+            BsonBinaryData binaryData;
+            if (BsonDefaults.GuidRepresentationMode == GuidRepresentationMode.V2)
```

Comment:
You'll see lots of if statements where the code that executes in V2 mode is unchanged.